src/api/database/misc.rst

```diff
@@ -78,47 +63,216 @@
+    :alt: Document Revision Tree 3
+
+    Document Revision Tree 3
+
+As a result of this purge operation, a new updated version of the document will
+be available in `_all_docs` and `_changes`, creating a new record in `_changes`.
+The database's `purge_seq` and `update_seq` will be increased.
+
+Internal Replication
+======================
+Purges are automatically replicated between replicas of the same database. Each
+database has an internal purge tree that stores a certain number of the most
+recent purges. This allows internal synchonization between replicas of the same
+database.
+
+External Replication
+======================
+Purge operations are not replicated to other external databases. External
+replication works by identifying source's document revisions that are missing on
+target, and copying these revisions from source to target. A purge operation
+completely purges revisions from a document's purge tree making external
+replication of purges not possible.
+
+    .. note::
+      If you are replicating between external databases and have
+      purged any documents and interested in having corresponding
+      purges in other databases as well, you should run purge
+      separately on each of the external databases.
+
 Updating Indexes
 ================
 
 The number of purges on a database is tracked using a purge sequence. This is
 used by the view indexer to optimize the updating of views that contain the
 purged documents.
 
-When the indexer identifies that the purge sequence on a database has changed,
-it compares the purge sequence of the database with that stored in the view
-index. If the difference between the stored sequence and database is sequence
-is only 1, then the indexer uses a cached list of the most recently purged
-documents, and then removes these documents from the index individually. This
-prevents completely rebuilding the index from scratch.
+Each internal database indexer, including the view indexer, keeps its own purge
+sequence. The purge sequence stored in index can be much smaller than the
+database's purge sequence up to the number of purge requests allowed to be
+stored in the purge trees of the database. Multiple purge requests can be
+processed by indexer without incurring rebuilt of the index. The index will be
+updated according to these purge requests.
+
+The index of document is based on winner of revision tree. Depending on which
+revision specified in purge request, there are different behaviors to update
+index:
+
+- if winner of revision tree is not specified in purge request, there is no
+  change on index record of this document;
+- if winner of revision tree is specified in purge request and there is still
+  revision left after purging, the index record of document will be built
+  according to new winner of revision tree;
+- if all revisions of document are specified in purge request, the index record
+  of document will be deleted. The document can't be searched out any more.
```

Comment:
- If all revisions of the document are specified in the purge request, the index record of the document will be deleted. The document will no longer be found in searches.